### ISSUE TITLE
[Problem]: core dumped after hours of usage

### ISSUE THREAD

**Issue by martinlindhe** [mikebrady/shairport-sync#1652]
### What happened?

Had it running and working for many hours, then it crashed. See log output

### Relevant log output

```shell
~  coredumpctl gdb 1373
           PID: 1373 (shairport-sync)
           UID: 1000 (m)
           GID: 1000 (m)
        Signal: 6 (ABRT)
     Timestamp: Sun 2023-03-26 15:15:42 CEST (1h 28min ago)
  Command Line: /usr/bin/shairport-sync --log-to-syslog
    Executable: /usr/bin/shairport-sync
 Control Group: /user.slice/user-1000.slice/user@1000.service/app.slice/shairport-sync.service
          Unit: user@1000.service
     User Unit: shairport-sync.service
         Slice: user-1000.slice
     Owner UID: 1000 (m)
       Boot ID: 0a88ff218b6e46de8e03bc07b5e9f013
    Machine ID: 936c40ba5fa94d6d83b01008264f1f4c
      Hostname: bork
       Storage: /var/lib/systemd/coredump/core.shairport-sync.1000.0a88ff218b6e46de8e03bc07b5e9f013.1373.1679836542000000.zst (present)
  Size on Disk: 694.1K
       Message: Process 1373 (shairport-sync) of user 1000 dumped core.

                Stack trace of thread 1404:
                #0  0x00007f55c86808ec n/a (libc.so.6 + 0x878ec)
                #1  0x00007f55c8631ea8 raise (libc.so.6 + 0x38ea8)
                #2  0x00007f55c861b53d abort (libc.so.6 + 0x2253d)
                #3  0x00007f55c8f2430f n/a (libpulse.so.0 + 0x1d30f)
                #4  0x00007f55c84fcd1f n/a (libpulsecommon-16.1.so + 0x48d1f)
                #5  0x00007f55c8f2ddd8 pa_mainloop_dispatch (libpulse.so.0 + 0x26dd8)
                #6  0x00007f55c8f2e43b pa_mainloop_iterate (libpulse.so.0 + 0x2743b)
                #7  0x00007f55c8f2e4e1 pa_mainloop_run (libpulse.so.0 + 0x274e1)
                #8  0x00007f55c8f3ec02 n/a (libpulse.so.0 + 0x37c02)
                #9  0x00007f55c850fc67 n/a (libpulsecommon-16.1.so + 0x5bc67)
                #10 0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #11 0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 143030:
                #0  0x00007f55c868c200 n/a (libc.so.6 + 0x93200)
                #1  0x00007f55c868ee63 __libc_free (libc.so.6 + 0x95e63)
                #2  0x0000562b7cf1280b n/a (shairport-sync + 0x880b)
                #3  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #4  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 143028:
                #0  0x00007f55c867b766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f55c86806a3 n/a (libc.so.6 + 0x876a3)
                #2  0x0000562b7cf398dd n/a (shairport-sync + 0x2f8dd)
                #3  0x0000562b7cf1aa8c n/a (shairport-sync + 0x10a8c)
                #4  0x0000562b7cf1e7ae n/a (shairport-sync + 0x147ae)
                #5  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #6  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 143029:
                #0  0x00007f55c86c5db5 clock_nanosleep (libc.so.6 + 0xccdb5)
                #1  0x00007f55c86ca987 __nanosleep (libc.so.6 + 0xd1987)
                #2  0x00007f55c86f6aa9 usleep (libc.so.6 + 0xfdaa9)
                #3  0x0000562b7cf215b3 n/a (shairport-sync + 0x175b3)
                #4  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1373:
                #0  0x00007f55c867b766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f55c86806a3 n/a (libc.so.6 + 0x876a3)
                #2  0x0000562b7cf16849 main (shairport-sync + 0xc849)
                #3  0x00007f55c861c790 n/a (libc.so.6 + 0x23790)
                #4  0x00007f55c861c84a __libc_start_main (libc.so.6 + 0x2384a)
                #5  0x0000562b7cf179f5 n/a (shairport-sync + 0xd9f5)

                Stack trace of thread 1418:
                #0  0x00007f55c867b766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f55c867df90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x0000562b7cf2547c n/a (shairport-sync + 0x1b47c)
                #3  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #4  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1413:
                #0  0x00007f55c867b766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f55c867df90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x0000562b7cf39a7c n/a (shairport-sync + 0x2fa7c)
                #3  0x0000562b7cf21c37 n/a (shairport-sync + 0x17c37)
                #4  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1414:
                #0  0x00007f55c867b766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f55c867df90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x0000562b7cf39a7c n/a (shairport-sync + 0x2fa7c)
                #3  0x0000562b7cf1b000 n/a (shairport-sync + 0x11000)
                #4  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1415:
                #0  0x00007f55c867b766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f55c867df90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x0000562b7cf39a7c n/a (shairport-sync + 0x2fa7c)
                #3  0x0000562b7cf19e6f n/a (shairport-sync + 0xfe6f)
                #4  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1416:
                #0  0x00007f55c867b766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f55c867df90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x0000562b7cf39a7c n/a (shairport-sync + 0x2fa7c)
                #3  0x0000562b7cf1b2bf n/a (shairport-sync + 0x112bf)
                #4  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1417:
                #0  0x00007f55c86c5db5 clock_nanosleep (libc.so.6 + 0xccdb5)
                #1  0x00007f55c86ca987 __nanosleep (libc.so.6 + 0xd1987)
                #2  0x00007f55c86ca8be sleep (libc.so.6 + 0xd18be)
                #3  0x0000562b7cf34351 n/a (shairport-sync + 0x2a351)
                #4  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1419:
                #0  0x00007f55c86f61cc __select (libc.so.6 + 0xfd1cc)
                #1  0x0000562b7cf1f814 n/a (shairport-sync + 0x15814)
                #2  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #3  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1421:
                #0  0x00007f55c86f39df __poll (libc.so.6 + 0xfa9df)
                #1  0x00007f55c8dd6a6c n/a (libavahi-common.so.3 + 0x7a6c)
                #2  0x00007f55c8dd65f1 avahi_simple_poll_run (libavahi-common.so.3 + 0x75f1)
                #3  0x00007f55c8dd67c9 avahi_simple_poll_iterate (libavahi-common.so.3 + 0x77c9)
                #4  0x00007f55c8dd69fe avahi_simple_poll_loop (libavahi-common.so.3 + 0x79fe)
                #5  0x00007f55c8dd6acb n/a (libavahi-common.so.3 + 0x7acb)
                #6  0x00007f55c867ebb5 n/a (libc.so.6 + 0x85bb5)
                #7  0x00007f55c8700d90 n/a (libc.so.6 + 0x107d90)
                ELF object binary architecture: AMD x86-64

GNU gdb (GDB) 13.1
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/shairport-sync...
Downloading separate debug info for /usr/bin/shairport-sync
(No debugging symbols found in /usr/bin/shairport-sync)

warning: Can't open file /memfd:pulseaudio (deleted) during file-backed mapping note processing
[New LWP 1404]
[New LWP 143030]
[New LWP 143028]
[New LWP 143029]
[New LWP 1373]
[New LWP 1418]
[New LWP 1413]
[New LWP 1414]
[New LWP 1415]
[New LWP 1416]
[New LWP 1417]
[New LWP 1419]
[New LWP 1421]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Core was generated by `/usr/bin/shairport-sync --log-to-syslog'.
Program terminated with signal SIGABRT, Aborted.
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
Downloading source file /usr/src/debug/glibc/glibc/nptl/pthread_kill.c
44            return INTERNAL_SYSCALL_ERROR_P (ret) ? INTERNAL_SYSCALL_ERRNO (ret) : 0;
[Current thread is 1 (Thread 0x7f55c3be16c0 (LWP 1404))]
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007f55c8680953 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007f55c8631ea8 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007f55c861b53d in __GI_abort () at abort.c:79
#4  0x00007f55c8f2430f in mainloop_defer_enable (e=<optimized out>, b=<optimized out>) at ../pulseaudio/src/pulse/mainloop.c:258
#5  0x00007f55c84fcd1f in do_pstream_read_write (p=0x7f55bc000c40) at ../pulseaudio/src/pulsecore/pstream.c:245
#6  0x00007f55c8f2ddd8 in dispatch_pollfds (m=0x562b7d8708a0) at ../pulseaudio/src/pulse/mainloop.c:676
#7  pa_mainloop_dispatch (m=m@entry=0x562b7d8708a0) at ../pulseaudio/src/pulse/mainloop.c:917
#8  0x00007f55c8f2e43b in pa_mainloop_iterate (m=m@entry=0x562b7d8708a0, block=block@entry=1, retval=retval@entry=0x0)
    at ../pulseaudio/src/pulse/mainloop.c:948
#9  0x00007f55c8f2e4e1 in pa_mainloop_run (m=0x562b7d8708a0, retval=0x0) at ../pulseaudio/src/pulse/mainloop.c:963
#10 0x00007f55c8f3ec02 in thread (userdata=0x562b7d870850) at ../pulseaudio/src/pulse/thread-mainloop.c:101
#11 0x00007f55c850fc67 in internal_thread_func (userdata=0x562b7d87c410) at ../pulseaudio/src/pulsecore/thread-posix.c:81
#12 0x00007f55c867ebb5 in start_thread (arg=<optimized out>) at pthread_create.c:444
#13 0x00007f55c8700d90 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
(gdb)
```

### Configuration Information.

         0.001214952 "shairport.c:1424" default metadata_pipename is "/tmp/shairport-sync-metadata".
         0.000065953 "shairport.c:2061" >> Display Config Start.
         0.007190192 "shairport.c:2061"
         0.000021155 "shairport.c:2061" From "uname -a":
         0.000005141 "shairport.c:2061"  Linux bork 6.2.8-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 22 Mar 2023 22:52:35 +0000 x86_64 GNU/Linux
         0.009569141 "shairport.c:2061"
         0.000004208 "shairport.c:2061" From /etc/os-release:
         0.000001972 "shairport.c:2061"  Arch Linux
         0.005557829 "shairport.c:2061"
         0.000005334 "shairport.c:2061" Shairport Sync Version String:
         0.000001792 "shairport.c:2061"  4.1.1-OpenSSL-Avahi-dns_sd-ALSA-jack-pa-stdout-pipe-soxr-metadata-mqtt-sysconfdir:/etc
         0.000003861 "shairport.c:2061"
         0.000002526 "shairport.c:2061" Command Line:
         0.000001594 "shairport.c:2061"  shairport-sync --displayConfig
         0.000005624 "shairport.c:2061"
         0.000002361 "shairport.c:2061" Configuration File:
         0.000003128 "shairport.c:2061"  /etc/shairport-sync.conf
         0.000002333 "shairport.c:2061"
         0.000023567 "shairport.c:2061" Configuration File Settings:
         0.000003168 "shairport.c:2061"  general :
         0.000002708 "shairport.c:2061"  {
         0.000002352 "shairport.c:2061"    name = "Stereo";
         0.000002978 "shairport.c:2061"    output_backend = "pa";
         0.000002579 "shairport.c:2061"  };
         0.000001555 "shairport.c:2061"  sessioncontrol :
         0.000001580 "shairport.c:2061"  {
         0.000002457 "shairport.c:2061"    allow_session_interruption = "yes";
         0.000003212 "shairport.c:2061"    session_timeout = 604800;
         0.000001574 "shairport.c:2061"  };
         0.000002425 "shairport.c:2061"  diagnostics :
         0.000001528 "shairport.c:2061"  {
         0.000002450 "shairport.c:2061"    log_output_to = "stdout";
         0.000001615 "shairport.c:2061"    statistics = "yes";
         0.000001487 "shairport.c:2061"    log_verbosity = 2;
         0.000003788 "shairport.c:2061"  };
         0.000002639 "shairport.c:2061"
         0.000001604 "shairport.c:2061" >> Display Config End.
         0.000001519 "shairport.c:2063" >> Goodbye!
         0.000004236 "shairport.c:1538" exit function called...
         0.000002973 "shairport.c:1574" Stopping DACP Monitor
         0.000001649 "shairport.c:1576" Stopping DACP Monitor Done
         0.000002470 "shairport.c:1580" Stopping metadata hub
         0.000002538 "shairport.c:1582" Stopping metadata done
         0.000002398 "shairport.c:1586" Stopping metadata
         0.000002385 "shairport.c:1588" Stopping metadata done
         0.000002718 "shairport.c:1590" Stopping the activity monitor.
         0.000002410 "shairport.c:1592" Stopping the activity monitor done.
         0.000004847 "shairport.c:1674" normal exit

### How did you install Shairport Sync?

A package manager (apt, apt install, yum, pkg, etc.)

### Check previous issues

- [X] Confirm

**Comment by mikebrady** [comment]
Thanks for this interesting log! It looks as if the PulseAudio subsystem aborted the app, but it’s not possible to say why. Was anything happening in the general environment?

**Comment by martinlindhe** [comment]
> Thanks for this interesting log! It looks as if the PulseAudio subsystem aborted the app, but it’s not possible to say why. Was anything happening in the general environment?

Nothing in particular afaik. The system was running for days, and I had been sending airplay audio to it from my apple tv for several hours hours before I noticed the airplay destination was no longer showing up on the Apple TV. Went to investigate and found the core dump.

I had shairport-sync crash before 1 or 2 times before earlier in the year, but it seems my system has scrubbed itself from older core dumps, so unfortunatley I cannot provide backtraces for those. But the symptom seems same/similar to me, that is- everything works for hours or even days, and then eventually a core dump.

**Comment by martinlindhe** [comment]
Had another crash, looks to be the same signature

```
coredumpctl gdb 1328
           PID: 1328 (shairport-sync)
           UID: 1000 (m)
           GID: 1000 (m)
        Signal: 6 (ABRT)
     Timestamp: Thu 2023-03-30 21:39:51 CEST (25min ago)
  Command Line: /usr/bin/shairport-sync --log-to-syslog
    Executable: /usr/bin/shairport-sync
 Control Group: /user.slice/user-1000.slice/user@1000.service/app.slice/shairport-sync.service
          Unit: user@1000.service
     User Unit: shairport-sync.service
         Slice: user-1000.slice
     Owner UID: 1000 (m)
       Boot ID: 261e9730e2a64c068233cbf0b2fed102
    Machine ID: 936c40ba5fa94d6d83b01008264f1f4c
      Hostname: bork
       Storage: /var/lib/systemd/coredump/core.shairport-sync.1000.261e9730e2a64c068233cbf0b2fed102.1328.1680205191000000.zst (present)
  Size on Disk: 1.4M
       Message: Process 1328 (shairport-sync) of user 1000 dumped core.

                Stack trace of thread 1358:
                #0  0x00007f00bc7b88ec n/a (libc.so.6 + 0x878ec)
                #1  0x00007f00bc769ea8 raise (libc.so.6 + 0x38ea8)
                #2  0x00007f00bc75353d abort (libc.so.6 + 0x2253d)
                #3  0x00007f00bd08c30f n/a (libpulse.so.0 + 0x1d30f)
                #4  0x00007f00bc652d1f n/a (libpulsecommon-16.1.so + 0x48d1f)
                #5  0x00007f00bd095dd8 pa_mainloop_dispatch (libpulse.so.0 + 0x26dd8)
                #6  0x00007f00bd09643b pa_mainloop_iterate (libpulse.so.0 + 0x2743b)
                #7  0x00007f00bd0964e1 pa_mainloop_run (libpulse.so.0 + 0x274e1)
                #8  0x00007f00bd0a6c02 n/a (libpulse.so.0 + 0x37c02)
                #9  0x00007f00bc665c67 n/a (libpulsecommon-16.1.so + 0x5bc67)
                #10 0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #11 0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1328:
                #0  0x00007f00bc7b3766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f00bc7b86a3 n/a (libc.so.6 + 0x876a3)
                #2  0x000056252e910849 main (shairport-sync + 0xc849)
                #3  0x00007f00bc754790 n/a (libc.so.6 + 0x23790)
                #4  0x00007f00bc75484a __libc_start_main (libc.so.6 + 0x2384a)
                #5  0x000056252e9119f5 n/a (shairport-sync + 0xd9f5)

                Stack trace of thread 82352:
                #0  0x00007f00bc7b3766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f00bc7b86a3 n/a (libc.so.6 + 0x876a3)
                #2  0x000056252e9338dd n/a (shairport-sync + 0x2f8dd)
                #3  0x000056252e914a8c n/a (shairport-sync + 0x10a8c)
                #4  0x000056252e9187ae n/a (shairport-sync + 0x147ae)
                #5  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #6  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1372:
                #0  0x00007f00bc7b3766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f00bc7b5f90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x000056252e91f47c n/a (shairport-sync + 0x1b47c)
                #3  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #4  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 82354:
                #0  0x00007f00bc7b6a9e n/a (libc.so.6 + 0x85a9e)
                #1  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1370:
                #0  0x00007f00bc7b3766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f00bc7b5f90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x000056252e933a7c n/a (shairport-sync + 0x2fa7c)
                #3  0x000056252e9152bf n/a (shairport-sync + 0x112bf)
                #4  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1368:
                #0  0x00007f00bc7b3766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f00bc7b5f90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x000056252e933a7c n/a (shairport-sync + 0x2fa7c)
                #3  0x000056252e915000 n/a (shairport-sync + 0x11000)
                #4  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1367:
                #0  0x00007f00bc7b3766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f00bc7b5f90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x000056252e933a7c n/a (shairport-sync + 0x2fa7c)
                #3  0x000056252e91bc37 n/a (shairport-sync + 0x17c37)
                #4  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1371:
                #0  0x00007f00bc7fddb5 clock_nanosleep (libc.so.6 + 0xccdb5)
                #1  0x00007f00bc802987 __nanosleep (libc.so.6 + 0xd1987)
                #2  0x00007f00bc8028be sleep (libc.so.6 + 0xd18be)
                #3  0x000056252e92e351 n/a (shairport-sync + 0x2a351)
                #4  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 82353:
                #0  0x00007f00bc7fddb5 clock_nanosleep (libc.so.6 + 0xccdb5)
                #1  0x00007f00bc802987 __nanosleep (libc.so.6 + 0xd1987)
                #2  0x00007f00bc82eaa9 usleep (libc.so.6 + 0xfdaa9)
                #3  0x000056252e91b5b3 n/a (shairport-sync + 0x175b3)
                #4  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1374:
                #0  0x00007f00bc82b9df __poll (libc.so.6 + 0xfa9df)
                #1  0x00007f00bcf3ea6c n/a (libavahi-common.so.3 + 0x7a6c)
                #2  0x00007f00bcf3e5f1 avahi_simple_poll_run (libavahi-common.so.3 + 0x75f1)
                #3  0x00007f00bcf3e7c9 avahi_simple_poll_iterate (libavahi-common.so.3 + 0x77c9)
                #4  0x00007f00bcf3e9fe avahi_simple_poll_loop (libavahi-common.so.3 + 0x79fe)
                #5  0x00007f00bcf3eacb n/a (libavahi-common.so.3 + 0x7acb)
                #6  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #7  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1373:
                #0  0x00007f00bc82e1cc __select (libc.so.6 + 0xfd1cc)
                #1  0x000056252e919814 n/a (shairport-sync + 0x15814)
                #2  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #3  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)

                Stack trace of thread 1369:
                #0  0x00007f00bc7b3766 n/a (libc.so.6 + 0x82766)
                #1  0x00007f00bc7b5f90 pthread_cond_wait (libc.so.6 + 0x84f90)
                #2  0x000056252e933a7c n/a (shairport-sync + 0x2fa7c)
                #3  0x000056252e913e6f n/a (shairport-sync + 0xfe6f)
                #4  0x00007f00bc7b6bb5 n/a (libc.so.6 + 0x85bb5)
                #5  0x00007f00bc838d90 n/a (libc.so.6 + 0x107d90)
                ELF object binary architecture: AMD x86-64

GNU gdb (GDB) 13.1
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/shairport-sync...
(No debugging symbols found in /usr/bin/shairport-sync)

warning: Can't open file /memfd:pulseaudio (deleted) during file-backed mapping note processing
[New LWP 1358]
[New LWP 1328]
[New LWP 82352]
[New LWP 1372]
[New LWP 82354]
[New LWP 1370]
[New LWP 1368]
[New LWP 1367]
[New LWP 1371]
[New LWP 82353]
[New LWP 1374]
[New LWP 1373]
[New LWP 1369]
Downloading separate debug info for /usr/lib/libvorbis.so.0
Downloading separate debug info for /usr/lib/libogg.so.0
Downloading separate debug info for /usr/lib/libXau.so.6
Downloading separate debug info for /usr/lib/libXdmcp.so.6
Downloading separate debug info for /usr/lib/libcap.so.2
Downloading separate debug info for /usr/lib/libgcrypt.so.20
--Type <RET> for more, q to quit, c to continue without paging--q
Quit
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007f00bc7b8953 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007f00bc769ea8 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007f00bc75353d in __GI_abort () at abort.c:79
#4  0x00007f00bd08c30f in mainloop_defer_enable (e=<optimized out>, b=<optimized out>) at ../pulseaudio/src/pulse/mainloop.c:258
#5  0x00007f00bc652d1f in do_pstream_read_write (p=0x7f00b0000c60) at ../pulseaudio/src/pulsecore/pstream.c:245
#6  0x00007f00bd095dd8 in dispatch_pollfds (m=0x56252fd848a0) at ../pulseaudio/src/pulse/mainloop.c:676
#7  pa_mainloop_dispatch (m=m@entry=0x56252fd848a0) at ../pulseaudio/src/pulse/mainloop.c:917
#8  0x00007f00bd09643b in pa_mainloop_iterate (m=m@entry=0x56252fd848a0, block=block@entry=1, retval=retval@entry=0x0)
    at ../pulseaudio/src/pulse/mainloop.c:948
#9  0x00007f00bd0964e1 in pa_mainloop_run (m=0x56252fd848a0, retval=0x0) at ../pulseaudio/src/pulse/mainloop.c:963
#10 0x00007f00bd0a6c02 in thread (userdata=0x56252fd84850) at ../pulseaudio/src/pulse/thread-mainloop.c:101
#11 0x00007f00bc665c67 in internal_thread_func (userdata=0x56252fd90410) at ../pulseaudio/src/pulsecore/thread-posix.c:81
#12 0x00007f00bc7b6bb5 in start_thread (arg=<optimized out>) at pthread_create.c:444
#13 0x00007f00bc838d90 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
```

**Comment by martinlindhe** [comment]
I have not had another crash now for quite some while, I guess it resolved with some system update (pipewire?).

**Comment by martinlindhe** [comment]
This keeps happening to me, but I mitigate it by auto restarting systemd service.

**Comment by mikebrady** [comment]
Thanks for this. Is it always the same version of Shairport Sync, or you updating as you go along?

**Comment by martinlindhe** [comment]
I currently run the stable version off arch linux, which is at 4.1.1. I noticed that the arch package has not yet been updated to the 4.2 release [1]

1: https://archlinux.org/packages/extra/x86_64/shairport-sync/

I have kept my system up to date otherwise (theres been some pipewire updates), but this issue has been going on.

If I run `coredumpctl gdb <pid>`, I get the same backtrace as in the first posts as far as I could tell.
